config.go

```diff
@@ -406,6 +418,51 @@ func hasConfig(cfg interface{}) bool {
 	return tf.NumField() != 0
 }
 
+// assignFieldMapping verifies that field mapping has been set once, but only
+// once (either in cfg or comp). Then if that is the case, assignFieldMapping
+// sets both fieldByName and fieldName in cfg.
+func assignFieldMapping(cfg *Config, comp Components) error {
+	// First, get inconsistent cases out of the way.
+	if len(cfg.Fields.Names) == 0 && comp.FieldByName == nil && comp.FieldName == nil {
+		return fmt.Errorf("field indexes/names have not been set")
+	}
+
+	if (comp.FieldByName == nil) != (comp.FieldName == nil) {
+		return fmt.Errorf("FieldByName and FieldName must be either both set or both unset")
+	}
+
+	if len(cfg.Fields.Names) != 0 && comp.FieldByName != nil && comp.FieldName != nil {
```

Comment:
shouldn't this be `a && (B || C)`?

Reply:
At this point we already know that `B == C` (they're either both true or both false, cf L430).

Reply:
ok, agree. But still I'm not sure why we have `comp.FieldByName != nil && comp.FieldName != nil` here.

I mean: or we check `a && (B || C)` (because `a && B && C` is incorrect) or we can remove the `&& comp.FieldName != nil` part, no?

I say **incorrect** because the only chance to hit the 3rd part of the `if`, (i.e. `&& comp.FieldName != nil`) is if the previous check (L430) is removed. But, in that case, `A && B && C` is a wrong check (`a==4, b==nil, c==&someptr` doesn't trigger the error)

Reply:
I thought about that, the last check is surely not needed, but I wanted to keep the symmetry between both functions checks because it seemed misleading to me to only check for one of them.
I think I found a solution to make the code more easily readable

Reply:
done